blog/2022-04-06-SDL-snowpark/2022-04-06-SDL-snowpark.md

```diff
@@ -0,0 +1,260 @@
+---
+title: Combine Spark and Snowpark to ingest and transform data in one pipeline
+description: An example to create one unified data pipeline that uses Spark to ingest data into Snowflake, and Snowpark to transform data inside Snowflake.
+slug: sdl-snowpark
+authors:
+  - name: Zach Kull
+    title: Data Expert
+    url: https://www.linkedin.com/in/zacharias-kull-94705886/
+tags: [Snowpark, Snowflake]
+hide_table_of_contents: false
+---
+
+This article shows how to create one unified data pipeline that uses Spark to ingest data into Snowflake, and Snowpark to transform data inside Snowflake.
+
+<!--truncate-->
+
+Recent developments in Smart Data Lake Builder (SDLB) included refactorings to integrate alternative execution engines to Spark.
+In particular [Snowpark](https://docs.snowflake.com/en/developer-guide/snowpark/index.html) integration was implemented, a Spark like DataFrame API for implementing transformations in Snowflake.
+
+Implementing transformations in Snowflake has big performance and cost benefits. And using a DataFrame API is much more powerful than coding in SQL, see also [Modern Data Stack: Which Place for Spark?](https://medium.com/towards-data-science/modern-data-stack-which-place-for-spark-8e10365a8772).
+
+Snowpark is good for transforming data inside Snowflake, but not all data might be located in Snowflake and suitable for Snowflake. 
+Here it is interesting to use Spark and its many connectors, in particular to ingest and export data.
+
+Combining Spark and Snowpark in a smart data pipeline using a DataFrame API would be the ideal solution.
+With the integration of Snowpark as engine in SDLB we created just that. 
+
+This blog post will show how to migrate our example data pipeline of the [Getting Started](../../docs/getting-started/setup) guide Part 1 to use Spark for ingestion and Snowpark for transformation.
+
+## Prerequisits

```

Comment:
I'm mostly fixing some typos.
Did you know that IntelliJ does spekk checking?
![image](https://user-images.githubusercontent.com/6410125/161817773-698661a3-852a-417a-a7ca-168e46a14e94.png)

Reply:
Mh, i was working in visual studio, but you're right, i should look for a spell checking plugin.